code/scripting/api/objs/ship_bank.cpp

```diff
@@ -409,6 +403,13 @@ ADE_VIRTVAR(FOFCooldown, l_WeaponBank, "number", "The FOF cooldown value. A valu
 			float reset_amount = (timestamp_until(bh->typeh.sw->last_primary_fire_stamp[bh->bank]) / 1000.0f) * wif->fof_reset_rate;
 			auto val = bh->typeh.sw->primary_bank_fof_cooldown[bh->bank] + reset_amount;
 			CLAMP(val, 0.0f, 1.0f);
+
+			if (ADE_SETTING_VAR) {
+				CLAMP(newValue, 0.0f, 1.0f);
+				bh->typeh.sw->primary_bank_fof_cooldown[bh->bank] = newValue;
```

Comment:
This should go at the top of the block because that way the code will properly handle the `reset_amount`.  Assigning `val = newValue` is then not needed.

Reply:
I'm not sure I follow, if it's being set then the reset_amount doesn't matter, it's going to be overriden anyway.

Reply:
Oh, so the value that is being set should be the grand total, without adding `reset_amount`?  Ok, that makes sense.

Although `reset_amount` will be added the very next time the value is retrieved.  Should the function account for that, perhaps by setting `newValue - reset_amount`?

Reply:
Okay